### PR TITLE
Simplifies functions in cluster.py.

### PR DESCRIPTION
**Pull request by jfinkels**
Replaces numbers like `1.0` with `1` after importing division from `__future__`, which should solve the problem that using the decimal points intended to solve.

### CODE REVIEW

File: networkx/algorithms/cluster.py [hunk 1]
```diff
@@ -352,20 +339,21 @@ def square_clustering(G, nodes=None):
...
+            degm = squares + 1
             if w in G[u]:
                 degm += 1
             potential += (len(G[u]) - degm) * (len(G[w]) - degm) + squares
         if potential > 0:
             clustering[v] /= potential
     if nodes in G: 
-        return list(clustering.values())[0] # return single value
+        # Return the value of the sole entry in the dictionary.
+        return arbitrary_element(clustering.values())
```
Comment (hagberg) [inline networkx/algorithms/cluster.py]: This reads kind of strange.  "arbitrary element" when there is only one element...

Comment (jfinkels) [inline networkx/algorithms/cluster.py]: I suppose `sole_element()` would read better. I could create it as an alias to `arbitrary_element`, and document that `sole_element()` has a pre-condition that its argument must have length one. In any case, `list(iterable)[0]` is less readable, I think, because the others indicate a bit more what's happening in plain English.

Comment (hagberg) [inline networkx/algorithms/cluster.py]: It's not that big of an issue.  The comment there is clear.

Comment (dschult) [inline networkx/algorithms/cluster.py]: How about ```return clustering[nodes]```?

Comment (hagberg) [inline networkx/algorithms/cluster.py]: Aww, so simple :smile:

Comment (jfinkels) [inline networkx/algorithms/cluster.py]: Good call, I will update this. Thanks.

Comment (dschult) [inline networkx/algorithms/cluster.py]: same handling of single element dict can work at line 253.

**Comment by jfinkels** [comment]
Updated due to style suggestions.
